Handle permission errors during plugin loading

syntaxsurge opened the pull request:
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

This pull request introduces a modification to the `load_plugins` function within `plugins.py` to gracefully handle `PermissionError` exceptions that occur during the plugin loading process. The modification ensures that `yt-dlp` can continue to run without crashing when it encounters permission issues, specifically when trying to access directories for which it does not have permissions.

This issue has been observed in environments where `yt-dlp` attempts to access restricted directories (e.g., `/root/yt_dlp_plugins`) and throws a `PermissionError`, leading to the program's termination. Such behavior has been reported when running `yt-dlp --version` in a virtual environment and similarly affects other operations, including starting FastAPI applications with Gunicorn where `yt-dlp` is a dependency.

**Example Error:**
```
PermissionError: [Errno 13] Permission denied: '/root/yt_dlp_plugins'
```

The proposed change wraps the plugin loading iteration in a `try-except` block to catch `PermissionError` exceptions, logs the error for debugging purposes, and allows the program to proceed by skipping the problematic plugin loading phase.

**Fixes:** Not applicable as this is a general improvement to error handling.

<details open><summary>Template</summary>

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [x] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details>

pukkandan commented:
Don't wrap the entire code block. Only `try except` where the error happens

pukkandan commented:
> ```shell
>   File "/home/jade/PycharmProjects/api_saas/venv/lib/python3.10/site-packages/yt_dlp/plugins.py", line 89, in search_locations
>     if candidate.is_dir():
> ```

The `try..except` is still too broad. Why not move it to where error actually happens? Like this

```diff
diff --git a/yt_dlp/plugins.py b/yt_dlp/plugins.py
index 6422c7a51..9f61bb6e0 100644
--- a/yt_dlp/plugins.py
+++ b/yt_dlp/plugins.py
@@ -86,11 +86,14 @@ def _get_package_paths(*root_paths, containing_folder='plugins'):
         parts = Path(*fullname.split('.'))
         for path in orderedSet(candidate_locations, lazy=True):
             candidate = path / parts
-            if candidate.is_dir():
-                yield candidate
-            elif path.suffix in ('.zip', '.egg', '.whl') and path.is_file():
-                if parts in dirs_in_zip(path):
+            try:
+                if candidate.is_dir():
                     yield candidate
+                elif path.suffix in ('.zip', '.egg', '.whl') and path.is_file():
+                    if parts in dirs_in_zip(path):
+                        yield candidate
+            except PermissionError as e:
+                write_string(f'Permission error while accessing modules in {e.filename}\n')

     def find_spec(self, fullname, path=None, target=None):
         if fullname not in self.packages:
```

syntaxsurge commented:
> > ```shell
> >   File "/home/jade/PycharmProjects/api_saas/venv/lib/python3.10/site-packages/yt_dlp/plugins.py", line 89, in search_locations
> >     if candidate.is_dir():
> > ```
> 
> The `try..except` is still too broad. Why not move it to where error actually happens? Like this
> 
> ```diff
> diff --git a/yt_dlp/plugins.py b/yt_dlp/plugins.py
> index 6422c7a51..9f61bb6e0 100644
> --- a/yt_dlp/plugins.py
> +++ b/yt_dlp/plugins.py
> @@ -86,11 +86,14 @@ def _get_package_paths(*root_paths, containing_folder='plugins'):
>          parts = Path(*fullname.split('.'))
>          for path in orderedSet(candidate_locations, lazy=True):
>              candidate = path / parts
> -            if candidate.is_dir():
> -                yield candidate
> -            elif path.suffix in ('.zip', '.egg', '.whl') and path.is_file():
> -                if parts in dirs_in_zip(path):
> +            try:
> +                if candidate.is_dir():
>                      yield candidate
> +                elif path.suffix in ('.zip', '.egg', '.whl') and path.is_file():
> +                    if parts in dirs_in_zip(path):
> +                        yield candidate
> +            except PermissionError as e:
> +                write_string(f'Permission error while accessing modules in {e.filename}\n')
> 
>      def find_spec(self, fullname, path=None, target=None):
>          if fullname not in self.packages:
> ```

Thank you for your guidance. I've implemented the change as you suggested. This should make the error handling more specific and informative, directly addressing the issue without unnecessarily broad exception catching. I appreciate your feedback and have committed the changes accordingly.

The output is now:
```bash
(venv) jade@jade-ASUS-TUF-Gaming-A15:~/PycharmProjects/api_saas/kubernetes$ yt-dlp --version
Permission error while accessing modules in /root/yt_dlp_plugins: [Errno 13] Permission denied: '/root/yt_dlp_plugins'
2023.12.30
```
